packages/@aws-cdk/region-info/README.md

```diff
@@ -0,0 +1,46 @@
+## AWS region information, such as service principal names
+### Usage
+Information pertaining to an AWS Region can be obtained using the `RegionInfo`
+class:
+
+```ts
+const dublin = RegionInfo.forRegion('eu-west-1');
+```
+
+### User-supplied region information
+The library comes pre-loaded with information about all publicly available AWS
+regions as of the release date of a particular version. It is possible this
+information is incomplete (or possibly incorrect). A user can provide their own
+region information by placing data in a JSON document at
+`~/.cdk/region-info.json`. The location of the file can be customized either by
```

Comment:
We shouldn't require that customization comes from a file, just allow people to pass it in in-memory. If they want to load it from a file, I am sure they'll figure out how to do that. I don't see value in implementing the file load in this library (and I do see value in being able to supply customizations at runtime through a strongly-typed interface, especially if it's an "escape hatch"). 

It also decreases the surface area (no environment variables, userDataPath and all that jazz).

Reply:
I somewhat disagree.

This is going to come in very handy when we use it in a build system that shall go unnamed.

Reply:
>  in a build system that shall go unnamed.

ha?

Reply:
Also, since deployment is outside the scope of your CDK application... I don't want you to have to change your CDK application's source code just because it happens to be deployed to `xy-unknown-5`

Reply:
I guess we can allow... both. I don't have an issue with that.

Reply:
Again, I don't see the huge value in being too opinionated about this. We don't know what the use cases are going to be really. This is a fringy edge case for the very rare situation where people would want to use a region we don't support and in that case, I suspect it will be much simpler for them to simply be able to specify those customizations in code rather than having to wrangle files around their home directory.

If we see a huge need in the future to add support for loading from files/home directory, we can always add that later.